addons/l10n_ke_edi_tremol/wizard/account_move_send.py

```diff
@@ -0,0 +1,36 @@
+from odoo import _, models, fields, api
+from odoo.exceptions import UserError
+
+
+class AccountMoveSend(models.Model):
+    _inherit = 'account.move.send'
+
+    l10n_ke_edi_warning_message = fields.Text(compute='_compute_l10n_ke_edi_warning_message')
+
+    @api.depends('move_ids')
+    def _compute_l10n_ke_edi_warning_message(self):
+        for wizard in self:
+            warning_moves = wizard.move_ids.filtered(lambda m: m.country_code == 'KE' and not m._l10n_ke_fiscal_device_details_filled())
+            if warning_moves:
+                wizard.l10n_ke_edi_warning_message = '\n'.join([
+                    _("The following documents have no details related to the fiscal device."),
+                    *(warning_moves.mapped('name'))
+                ])
+            else:
+                wizard.l10n_ke_edi_warning_message = False
+
+    def _hook_invoice_document_before_pdf_report_render(self, invoice, invoice_data):
+        # EXTENDS account
+        super()._hook_invoice_document_before_pdf_report_render(invoice, invoice_data)
+        if invoice.country_code == 'KE' and not invoice._l10n_ke_fiscal_device_details_filled():
+            invoice_data['error'] = _(
+                "This document does not have details related to the fiscal device, a proforma invoice will be used."
+            )
+
+    def action_send_and_print(self, from_cron=False, allow_fallback_pdf=False):
+        # EXTENDS account - prevent Send & Print if KE invoices aren't validated and no fallback is allowed.
+        self.ensure_one()
+        if not allow_fallback_pdf \
```

Comment:
@clbr-odoo 
I know I'm a bit too late, but how are we supposed to have `allow_fallback_pdf = True` here?

Reply:
For example it will be True if called from the e-commerce, which is a flow we don't want to block.
If used directly from Send & Print on invoices Laurent decided that it's better to just raise and not allow the proforma.

Reply:
Alright, thanks for the info